indico/modules/events/abstracts/operations.py

```diff
@@ -39,15 +39,30 @@
 from indico.util.fs import secure_filename
 
 
+def _update_tracks(abstract, tracks, only_reviewed_for=False):
+    edit_track_mode = abstract.edit_track_mode
+    if edit_track_mode == EditTrackMode.none:
+        return
+
+    if tracks is None:
+        tracks = set()
+    elif isinstance(tracks, Track):
+        tracks = {tracks}
+
+    if edit_track_mode == EditTrackMode.both:
```

Comment:
Same here, I would be expecting the operation primitives to simply do things rather than checking states.